File: java/test/com/github/bazel_contrib/contrib_rules_jvm/javaparser/generators/workspace/com/gazelle/java/javaparser/generators/ParameterExportingClass.java
package com.github.bazel_contrib.contrib_rules_jvm.javaparser.generators.workspace.com.gazelle.java.javaparser.generators;

import example.external.ConstructorParam;
import example.external.PackageParam;
import example.external.ParameterizedArg;
import example.external.ParameterizedParam;
import example.external.PrivateParam;
import example.external.ProtectedParam;
import example.external.PublicParam;
import example.external.VarargParam;
import example.external.WildcardBound;

public class ParameterExportingClass {
  public ParameterExportingClass(ConstructorParam param) {}

  void acceptsPackage(PackageParam param) {}

  private void acceptsPrivate(PrivateParam param) {}

  protected void acceptsProtected(ProtectedParam param) {}

  public void acceptsPublic(PublicParam param) {}

  public void acceptsVarargs(VarargParam... params) {}

  public void acceptsParameterized(ParameterizedParam<ParameterizedArg> param) {}

  public void acceptsWildcard(ParameterizedParam<? extends WildcardBound> param) {}
}

File: java/test/com/github/bazel_contrib/contrib_rules_jvm/javaparser/generators/workspace/com/gazelle/kotlin/javaparser/generators/ParameterExportingClass.kt
package workspace.com.gazelle.kotlin.javaparser.generators

import example.external.ConstructorParam
import example.external.DataClassProperty
import example.external.FunctionTypeParam
import example.external.FunctionTypeReturn
import example.external.GenericArg
import example.external.GenericOuter
import example.external.InternalParam
import example.external.NullableParam
import example.external.PrivateConstructorParam
import example.external.PrivateParam
import example.external.ProtectedParam
import example.external.PublicParam
import example.external.SecondaryConstructorParam
import example.external.VarargParam

class ParameterExportingClass(position: ConstructorParam) {

  constructor(alt: SecondaryConstructorParam) : this(ConstructorParam())

  fun acceptsPublic(param: PublicParam) {}

  internal fun acceptsInternal(param: InternalParam) {}

  protected fun acceptsProtected(param: ProtectedParam) {}

  private fun acceptsPrivate(param: PrivateParam) {}

  fun acceptsVarargs(vararg params: VarargParam) {}

  fun acceptsNullable(param: NullableParam?) {}

  fun acceptsParameterized(param: GenericOuter<GenericArg>) {}

  fun acceptsFunctionType(callback: (FunctionTypeParam) -> FunctionTypeReturn) {}
}

data class ParameterDataClass(val prop: DataClassProperty)

class PrivateConstructorClass private constructor(param: PrivateConstructorParam)
